web/modules/custom/acceptance/src/Form/DeliverDigestMessagesForm.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\joinup_acceptance\Form;
+
+use Drupal\Core\Form\FormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Queue\QueueFactory;
+use Drupal\Core\Queue\QueueWorkerManagerInterface;
+use Drupal\Core\Queue\RequeueException;
+use Drupal\Core\Queue\SuspendQueueException;
+use Drupal\message_digest\Traits\MessageDigestTrait;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Form allowing to deliver digest messages.
+ */
+class DeliverDigestMessagesForm extends FormBase {
+
+  use MessageDigestTrait;
+
+  /**
+   * The queue service.
+   *
+   * @var \Drupal\Core\Queue\QueueFactory
+   */
+  protected $queueFactory;
+
+  /**
+   * The queue manager.
+   *
+   * @var \Drupal\Core\Queue\QueueWorkerManagerInterface
+   */
+  protected $queueManager;
+
+  /**
+   * Constructs a new form instance.
+   *
+   * @param \Drupal\Core\Queue\QueueFactory $queue_factory
+   *   The queue service.
+   * @param \Drupal\Core\Queue\QueueWorkerManagerInterface $queue_manager
+   *   The queue manager.
+   */
+  public function __construct(QueueFactory $queue_factory, QueueWorkerManagerInterface $queue_manager) {
+    $this->queueFactory = $queue_factory;
+    $this->queueManager = $queue_manager;
+
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container): self {
+    return new static(
+      $container->get('queue'),
+      $container->get('plugin.manager.queue_worker')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state): array {
+    $form['deliver'] = [
+      '#type' => 'submit',
+      '#value' => $this->t('Deliver message digests'),
+      '#submit' => [[$this, 'deliverDigestMessages']],
```

Comment:
Is there a reason to generate an explicit submit handler instead of using the default `submitForm`?

Reply:
Yes. This form is open to support also other _acceptance tasks_ in the future. So, I've anticipated that new buttons or form elements might be added, with a different functionality. Thus, I didn't want to use the form submit for this particular action. Each future button or form element will add their submit logic, separately, in their submit callbacks.

True, that the form & block & permissions are telling about message delivery. That I'll change to reflect the idea of generic _acceptance tasks_ UI.